File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.cpp
#include <string>
#include <vector>

using namespace std;

int getCount(int x) {
    int cnt = 0;
    
    for (int i = 1; i * i < x + 1; i++) {
        if (x % i == 0) {
            if (i == x / i) {
                cnt += 1;
            } else {
                cnt += 2;
            }
        }
    }
    return cnt;
}

int solution(int number, int limit, int power) {
    int answer = 0;
    
    for (int i = 1; i <= number; i++) {
        int c = getCount(i);
        answer += (c > limit) ? power : c;
    }
    return answer;
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def getCount(x):
    
    cnt = 0;
    for i in range(1, int(x ** 0.5) + 1):
        if (x % i == 0):
            if (i == x // i):
                cnt += 1
            else:
                cnt += 2
    
    return cnt

def solution(number, limit, power):
    answer = 0
    
    for n in range(1, number + 1):
        c = getCount(n)
        # print(c)
        answer += c if (c <= limit) else power
    
    return answer